demo.py

```diff
@@ -0,0 +1,214 @@
+            run_name=run_name,
+            pipeline="Sage-Bionetworks-Workflows/nf-synstage",
+            revision="main",
+            profiles=["sage"],
+            params={
+                "input": self.samplesheet,
+            },
+            workspace_secrets=["SYNAPSE_AUTH_TOKEN"],
+        )
+
+    def rnaseq_info(self) -> LaunchInfo:
+        """Generate LaunchInfo for nf-core/rnaseq."""
+        run_name = self.get_run_name("rnaseq")
+        ref_prefix = "https://raw.githubusercontent.com/nf-core/test-datasets/rnaseq3"
+
+        nextflow_config = """
+        process {
+            withName: 'RSEM_PREPAREREFERENCE_GENOME' {
+                ext.args2 = "--genomeSAindexNbases 7"
+            }
+        }
+        """
+
+        return LaunchInfo(
+            run_name=run_name,
+            pipeline="nf-core/rnaseq",
+            revision="3.11.2",
+            profiles=["sage"],
+            nextflow_config=dedent(nextflow_config),
+            params={
+                "input": self.get_staged_samplesheet(),
+                "outdir": f"{OUTDIR_PREFIX}/{run_name}",
+                "fasta": f"{ref_prefix}/reference/genome.fasta",
+                "gtf": f"{ref_prefix}/reference/genes.gtf.gz",
+                "gff": f"{ref_prefix}/reference/genes.gff.gz",
+                "transcript_fasta": f"{ref_prefix}/reference/transcriptome.fasta",
+                "additional_fasta": f"{ref_prefix}/reference/gfp.fa.gz",
+                "bbsplit_fasta_list": f"{ref_prefix}/reference/bbsplit_fasta_list.txt",
+                "hisat2_index": f"{ref_prefix}/reference/hisat2.tar.gz",
+                "salmon_index": f"{ref_prefix}/reference/salmon.tar.gz",
+                "rsem_index": f"{ref_prefix}/reference/rsem.tar.gz",
+                "skip_bbsplit": False,
+                "pseudo_aligner": "salmon",
+                "umitools_bc_pattern": "NNNN",
+            },
+        )
+
+    def synindex_info(self) -> LaunchInfo:
+        """Generate LaunchInfo for nf-synindex."""
+        rnaseq_info = self.rnaseq_info()
+        assert rnaseq_info.params
+
+        return LaunchInfo(
+            run_name=self.get_run_name("synindex"),
+            pipeline="Sage-Bionetworks-Workflows/nf-synindex",
+            revision="main",
+            profiles=["sage"],
+            params={
+                "s3_prefix": rnaseq_info.params["outdir"],
+                "parent_id": self.output_folder,
+            },
+            workspace_secrets=["SYNAPSE_AUTH_TOKEN"],
+        )
+
+
+class TowerRnaseqFlow(FlowSpec):
+    """Flow for processing RNA-seq dataset on Tower."""
+
+    tower = NextflowTowerOps()
+    synapse = SynapseOps()
+    s3 = s3fs.S3FileSystem()
+
+    # Synapse ID for a YAML file that describes an RNA-seq dataset
+    # Here's an example:
+    #   id: my_test_dataset
+    #   samplesheet: syn51514475
+    #   output_folder: syn51514559
+    dataset_id = Parameter("dataset_id", help="Dataset Synapse ID", type=str)
```

Comment:
How would this look with multiple datasets? Also I am fine with this, but here you are defining a dataset identifier to be a yaml file that has id, output, and samplesheet.

Reply:
We could have a YAML file with a list of datasets ([example](https://github.com/Sage-Bionetworks-Workflows/ntap-add5-scripts/blob/main/workflows-467.yaml)), and we can run the flow for each dataset. In Metaflow, there's a `foreach` [feature](https://docs.metaflow.org/metaflow/basics#foreach) for this purpose. In Airflow, there's [dynamic task mapping](https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/dynamic-task-mapping.html). 

I'm just not sure if we want to add this complexity. My preference is to keep the DAG for one dataset, which gets triggered manually. Separately, a scheduled DAG monitors something like a File View and uses Workflow Triggers to execute the single-dataset DAG.

Reply:
Not an issue, more of just a concept which also exists in Synapse